libs/test/SmartSyncTest/src/com/salesforce/androidsdk/smartsync/target/ParentChildrenSyncTest.java

```diff
@@ -701,89 +678,800 @@ public void testCleanResyncGhostsForParentChildrenTarget() throws Exception {
+     * @throws Exception
+     */
+    public void testSyncUpLocallyDeletedParentDeletedChildRemotelyUpdatedParent() throws Exception {
+        trySyncUpsWithVariousChanges(2, 2, Change.DELETE, Change.UPDATE, Change.DELETE, Change.NONE);
+    }
+
+    /**
+     * Sync up with locally deleted parent deleted child remotely deleted parent
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyDeletedParentDeletedChildRemotelyDeletedParent() throws Exception {
+        trySyncUpsWithVariousChanges(2, 2, Change.DELETE, Change.DELETE, Change.DELETE, Change.NONE);
+    }
 
-            // Checking contacts
-            checkDbExist(CONTACTS_SOUP, contactIdsOfFirstAccount, Constants.ID);
-            checkDbExist(CONTACTS_SOUP, contactIdsOfThirdAccount, Constants.ID);
-            switch (relationshipType) {
-                case MASTER_DETAIL:
-                    checkDbDeleted(CONTACTS_SOUP, contactIdsOfSecondAccount, Constants.ID);
+    /**
+     * Sync up with locally updated parent and no children
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyUpdatedParentNoChildren() throws Exception {
+        trySyncUpsWithVariousChanges(2, 0, Change.UPDATE, Change.NONE, Change.NONE, Change.NONE);
+    }
+
+    /**
+     * Sync up with locally updated parent remotely updated parent and no children
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyUpdatedParentRemotelyUpdatedParentNoChildren() throws Exception {
+        trySyncUpsWithVariousChanges(2, 0, Change.UPDATE, Change.UPDATE, Change.NONE, Change.NONE);
+    }
+
+    /**
+     * Sync up with locally updated parent remotely deleted parent and no children
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyUpdatedParentRemotelyDeletedParentNoChildren() throws Exception {
+        trySyncUpsWithVariousChanges(2, 0, Change.UPDATE, Change.DELETE, Change.NONE, Change.NONE);
+    }
+
+    /**
+     * Sync up with locally deleted parent and no children
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyDeletedParentNoChildren() throws Exception {
+        trySyncUpsWithVariousChanges(2, 0, Change.DELETE, Change.NONE, Change.NONE, Change.NONE);
+    }
+
+    /**
+     * Sync up with locally deleted parent remotely updated parent and no children
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyDeletedParentRemotelyUpdatedParentNoChildren() throws Exception {
+        trySyncUpsWithVariousChanges(2, 0, Change.DELETE, Change.UPDATE, Change.NONE, Change.NONE);
+    }
+
+    /**
+     * Sync up with locally deleted parent remotely deleted parent and no children
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyDeletedParentRemotelyDeletedParentNoChildren() throws Exception {
+        trySyncUpsWithVariousChanges(2, 0, Change.DELETE, Change.DELETE, Change.NONE, Change.NONE);
+    }
+
+    /**
+     * Helper for various sync up test
```

Comment:
Helper method for testing sync up with locally and/or remotely updated/deleted parent and/or child.
See method comments for details.